### PR TITLE
Management Doc for Projections

### LINKED ISSUE #264: Create gRPC projections management section

### PR DESCRIPTION
**Pull request by ylorph**
Management operations for projections 
Samples 
- [x] .NEt
- [ ] JAva
- [ ] JavaScript
- [ ] TypeScript
- [ ] Rust has no maganagemnt for projection yet at all. not included in the doc
- [ ] Go : wait for 1.0 release ?

! the projections samples still need to be adapted ( that''s in the README.md file ) 
Plannning to do the gRPC .net one.
We need to think how we're going to have it for the for the different  stacks.
Samples in client repo
- [x] .Net

### CODE REVIEW

**Comment by oskardudycz** [comment]
Personally, I'd prefer more context in docs, as I can get API reference in IDE. However, that's just my perspective. If we want to take it gradually then I'm good, but it would be worth knowing the @mat-mcloughlin and @alexeyzimarev thoughts.

**Comment by mat-mcloughlin** [comment]
Yeah, it would be nice to actually see samples of how to use each of the functions and what the various options mean

**Comment by ylorph** [comment]
rebased it , the build was failing for _some_ unrelated reason , since it works on main rebased it

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
For a detailed explanation of projections, see the [server documentation](/server/v21.6/docs/projections/).
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
The various gRPC client APIs include dedicated clients that allow you to manage projections.
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You can find the full sample code from these documentation on the respective [clients repositories](https://github.com/eventStore/?q=client)
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:16]:
```suggestion
Install the client SDK package to your project.

<xode-group>
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **FIX:** We should split it into JavaScript and TypeScript code blocks as in the other gRPC docs.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **FIX:** This snippet is incorrect. Java client has `EventStoreDBProjectionManagementClient`

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
:::warning
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **FIX:** This is incorrect, as Rust does not support projections and when it's published, then it'll use a dedicated client `EventStoreDBProjectionManagementClient`.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:180]: **SUGGESTION:** It'd be good to explain briefly what's the projection subsystem and why would you like to restart it.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **FIX:** This and other TypeScript sample is not aligned with .NET snippets convention. This shows API reference, but should show sample:
```
await client.restartSubsystem()
```
Plus, those snippets should be placed in the NodeJS client (and split into TypeScript and JavaScript).

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:217]: **SUGGESTION:** Would be good to add some context to this operation (the same apply to the methods below).

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You can only enable an existing projection. When you try to enable a non-existing projection, you'll get an error: 
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **FIX:** It'd be good to either use `TODO` or `// This is currently not available in the X client` to keep the consistency.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **SUGGESTION:** It might be worth explaining how to get access and/or link to the access management docs.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You can only disable an existing projection. When you try to disable a non-existing projection, you'll get an error:
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
The .Net clients before version 21.6 had an incorrect behaviour: they _were_ saving the checkpoint. 
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:289]: The same suggestion about giving context as in https://github.com/EventStore/documentation/pull/404/files#r689298658.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:366]: The same suggestion about giving context as in https://github.com/EventStore/documentation/pull/404/files#r689298658.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:404]: The same suggestion about giving context as in https://github.com/EventStore/documentation/pull/404/files#r689298658.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You can only abort an existing projection. When you try to abort a non-existing projection, you'll get an error:
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:447]: Also, are other methods idempotent or only `Abort`?**CONCERN:** I think that part about calling abort multiple times would be good to put in the docs instead of a snippet. Also, are other methods idempotent or only `Abort`? If it's specific to the abort method, then it's worth keeping it as it is, but if it's in general, then it might be worth adding a general note or to the other specific methods.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:75]: **SUGGESTION:** I think we should remove the wrapper method for the client creation in the snippet. We do not have such wrappers in other places.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **SUGGESTION:** I think that this should be the first snippet, as we should try to have a learning path. If I were looking for those docs then I'd at first want to learn how to create a projection, then how to run it (enable) then other methods like disable, abort and delete.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **CONCERN:** What does it mean `that runs until the end of the log and then continues running`? It'd be good to explain briefly what the created javascript code does and link to the JS syntax docs.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You can only update an existing projection. When you try to update a non-existing projection, you'll get an error:
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **SUGGESTION:** Would be good to add some context to this operation.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:547]: **SUGGESTION:** Would be good to add some context to this operation.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:765]: **SUGGESTION:** Would be good to add some context to this operation. Especially what's the Result and partition.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:728]: **SUGGESTION:** Would be good to add some context to this operation. Especially what's the state in fact.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: **SUGGESTION:** Would be good to add some context to this operation.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md:652]: **SUGGESTION:** Would be good to add some context to this operation. Especially what's the Result and partition. Plus, it might be worth only mentioning listing continuous projections, as listing all will be disabled soon (via RFC).

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]: Similar comment as https://github.com/EventStore/documentation/pull/404/files#r689297932 regarding the API Reference vs snippets.

**Review by oskardudycz — CHANGES_REQUESTED** [review]
See the comments above.

File: docs/clients/grpc/projections/README.md
Comment (mat-mcloughlin) [reply docs/clients/grpc/projections/README.md]: @ylorph agreed

**Comment by mat-mcloughlin** [comment]
@ylorph Oskar has done a much better job of reviewing this than me, can you please take a look at and address the comments before merging please

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md]: P.S. I'm not sure if that's correct, as 21.6 .NET gRPC client is not yet released (https://www.nuget.org/packages/EventStore.Client.Grpc.Streams)

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:180]: that's in the server doc if I'm not mistaken .
Mistaken , it's not there .
Dunno why it needs restarting .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: I'm expecting the respective library maintainers to add samples for the respective clients.
we can't do it all for them

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:115]: we have no docs on ACL .

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md]: I agree. However, we did that in the past. If we don't want to do it this time, I suggest leaving it as TODO (accordingly to the, e.g. Java ones). This inconsistency, in my opinion, does not look right.

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:217]: What context ? it enables a projection

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: move closer to the top

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: as above, as noted in the original PR description, we noeed to decide if we continue to create those or not for the eng. team

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: as above , and in PR description since the beginning, we need to decide if we continue to do that instead of the Eng. team

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: as above, and notes in the original PR description, we need a decision if we keep on doing this instead of the eng. team

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:765]: Result is what was written in the javascript passed, this is explained in the server doc .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:728]: That is explained in the server doc linked at the very top of the doc.

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md:652]: well we have user projection & system projection , if the RFC does not allow that , we have a problem .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as aboce commens , this needs a decision

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as aboce, this need a decsion

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as above, this needs a decision .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as above, this needs a decsion .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as aboce, this needs a decision

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as above, this needs a decision .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: same as aboce , this needs a decision .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: the details are explained at the end .

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md:217]: It would be good to explain what enabling/disabling means in this context. What are the consequences.

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: removed and added placeholder, we need to decisde if we still provide those samples ourselves or let the eng. team do it .

File: docs/clients/grpc/projections/README.md
Comment (ylorph) [reply docs/clients/grpc/projections/README.md]: removed

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md:180]: We should give at least basic information, so people don't need to jump back and forth to the docs. It may be a single sentence, but giving context, what are the consequences, etc. The same applies to other methods.

Comment (ylorph) [reply docs/clients/grpc/projections/README.md:180]: > Mistaken , it's not there .
> Dunno why it needs restarting .

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
<xode-block title="TypeScript">
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
<xode-block title="TypeScript">
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [inline docs/clients/grpc/projections/README.md]:
```suggestion
You must have access to a projection to enable it, see the [ACL documentation](/server/v21.6/docs/security/acl.html#stream-acl)
```

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md:652]: It seems that [`ListContinuous` should be renamed to `List`](https://github.com/EventStore/home/blob/master/RFCs/021%20-%20gRPC%20Projections.md)
See: 
obraz

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md:765]: I think that at least a brief explanation would be good to have, e.g. is it the last projection event?

File: docs/clients/grpc/projections/README.md
Comment (oskardudycz) [reply docs/clients/grpc/projections/README.md:728]: Yup, but at least a sentence what's the state, in fact, would be good.

**Review by oskardudycz — APPROVED** [review]
@ylorph, some comments are still left, but as they are non-blocking, so I'm approving.

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| `Version`                             | This is used internally, the version is increased when the projection is edited or reset |
```

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| `Epoch`                               | This is used internally, the epoch is increased when the projection is reset |
```

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| Subscribed | The projection has successfully subscribed to its readers, this happens while the projection is starting |
```

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| `WritesInProgress`                    | The number of write requests to emitted streams currently in progress, these writes can be batches of events |
```

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| `ReadsInProgress`                     | The number of read requests currently in progress |
```

File: docs/clients/grpc/projections/README.md
Comment (hayley-jean) [inline docs/clients/grpc/projections/README.md]:
```suggestion
| `WritePendingEventsBeforeCheckpoint`  | The number of events waiting to be appended to emitted streams before the pending checkpoint can be written |
```

**Pull request merged**
